File: demo-app-objc/CloudXObjCRemotePods/Views/Ad/InitInternalViewController.m
#import "InitInternalViewController.h"
#import "CLXDemoConfigManager.h"
#import "BaseAdViewController.h"
#import "DemoAppLogger.h"
#import <CloudXCore/CloudXCore.h>

@interface InitInternalViewController ()
@property (nonatomic, assign) BOOL isSDKInitialized;
@property (nonatomic, strong) UIButton *initializationButton;
@end

@implementation InitInternalViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"ObjC Demo";
    
    [self setupInitializationButton];
    
    self.isSDKInitialized = NO;
    [self updateStatusUIWithState:AdStateNoAd];
}

- (void)setupShowLogsButton {
    // Do nothing - no show logs button for InitInternalViewController
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self.navigationController setNavigationBarHidden:NO animated:animated];
}

- (void)setupInitializationButton {
    // Create initialization button
    self.initializationButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.initializationButton setTitle:@"Initialize SDK" forState:UIControlStateNormal];
    [self.initializationButton addTarget:self action:@selector(initializeSDK) forControlEvents:UIControlEventTouchUpInside];
    
    // Style the button
    self.initializationButton.backgroundColor = [UIColor systemBlueColor];
    [self.initializationButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    self.initializationButton.layer.cornerRadius = 8;
    self.initializationButton.titleLabel.font = [UIFont boldSystemFontOfSize:16];
    self.initializationButton.translatesAutoresizingMaskIntoConstraints = NO;
    
    [self.view addSubview:self.initializationButton];
    
    // Add constraints
    [NSLayoutConstraint activateConstraints:@[
        [self.initializationButton.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
        [self.initializationButton.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor],
        [self.initializationButton.widthAnchor constraintEqualToConstant:200],
        [self.initializationButton.heightAnchor constraintEqualToConstant:44]
    ]];
}

- (void)initializeSDK {
    if (self.isSDKInitialized) {
        [self showAlertWithTitle:@"SDK Already Initialized" message:@"The SDK is already initialized."];
        return;
    }
    
    [self updateStatusUIWithState:AdStateLoading];
    self.initializationButton.enabled = NO;
    
    CLXDemoConfig *config = [[CLXDemoConfigManager sharedManager] currentConfig];
    
    [[DemoAppLogger sharedInstance] logMessage:@"Initializing SDK"];
    
    // Set hashed user ID before initialization if provided
    if (config.hashedUserId.length > 0) {
        [[CloudXCore shared] setHashedUserID:config.hashedUserId];
    }
    
    // Use standard CloudXCore initialization
    // Production demo app - use testMode:NO for real ads
    [[CloudXCore shared] initializeSDKWithAppKey:config.appKey 
                                testMode:NO
                                completion:^(BOOL success, CLXError * _Nullable error) {
        dispatch_async(dispatch_get_main_queue(), ^{
            if (success) {
                [[DemoAppLogger sharedInstance] logMessage:@"✅ SDK initialized successfully"];
                self.isSDKInitialized = YES;
                [self updateStatusUIWithState:AdStateReady];
                [[NSNotificationCenter defaultCenter] postNotificationName:@"cloudXSDKInitialized" object:nil];
            } else {
                NSString *errorMessage = error ? error.localizedDescription : @"Unknown error occurred";
                [self showAlertWithTitle:@"SDK Init Failed" message:errorMessage];
                [self updateStatusUIWithState:AdStateNoAd];
                self.initializationButton.enabled = YES;
                [[DemoAppLogger sharedInstance] logMessage:[NSString stringWithFormat:@"❌ SDK init failed: %@", errorMessage]];
            }
        });
    }];
}

@end

File: demo-app-swift/CloudXSwiftRemotePods/Views/Ad/InitInternalViewController.swift
import UIKit
import CloudXCore

class InitInternalViewController: BaseAdViewController {
    
    private var isSDKInitialized: Bool = false
    private var initButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Swift Demo"
        
        setupInitButton()
        
        // SDK initialization state tracked internally
        isSDKInitialized = false
        updateStatusUI(state: .noAd)
    }
    
    // Override to prevent show logs button from appearing in InitInternalViewController
    override func setupShowLogsButton() {
        // Do nothing - no show logs button for InitInternalViewController
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        navigationController?.setNavigationBarHidden(false, animated: animated)
    }
    
    private func setupInitButton() {
        // Create init button
        initButton = UIButton(type: .system)
        initButton.setTitle("Init SDK", for: .normal)
        initButton.addTarget(self, action: #selector(initializeSDK), for: .touchUpInside)
        
        // Style the button
        initButton.backgroundColor = .systemBlue
        initButton.tintColor = .white
        initButton.layer.cornerRadius = 8
        initButton.titleLabel?.font = .boldSystemFont(ofSize: 16)
        initButton.translatesAutoresizingMaskIntoConstraints = false
        
        view.addSubview(initButton)
        
        // Add constraints
        NSLayoutConstraint.activate([
            initButton.centerXAnchor.constraint(equalTo: view.centerXAnchor),
            initButton.centerYAnchor.constraint(equalTo: view.centerYAnchor),
            initButton.widthAnchor.constraint(equalToConstant: 200),
            initButton.heightAnchor.constraint(equalToConstant: 44)
        ])
    }
    
    @objc private func initializeSDK() {
        if isSDKInitialized {
            showAlert(title: "SDK Already Initialized", message: "The SDK is already initialized.")
            return
        }
        
        updateStatusUI(state: .loading)
        initButton.isEnabled = false
        
        let config = CLXDemoConfigManager.sharedManager.currentConfig
        
        DemoAppLogger.sharedInstance.logMessage("Initializing SDK")
        
        // Set hashed user ID before initialization if provided
        if !config.hashedUserId.isEmpty {
            CloudXCore.shared.setHashedUserID(config.hashedUserId)
        }
        
        // Use standard CloudXCore initialization
        CloudXCore.shared.initializeSDK(appKey: config.appKey) { [weak self] success, error in
            guard let self = self else { return }
            
            DispatchQueue.main.async {
                if success {
                    DemoAppLogger.sharedInstance.logMessage("✅ SDK initialized successfully")
                    self.isSDKInitialized = true
                    self.updateStatusUI(state: .ready)
                    NotificationCenter.default.post(name: .sdkInitialized, object: nil)
                } else {
                    let errorMessage = error?.localizedDescription ?? "Unknown error occurred"
                    self.showAlert(title: "SDK Init Failed", message: errorMessage)
                    self.updateStatusUI(state: .noAd)
                    self.initButton.isEnabled = true
                    DemoAppLogger.sharedInstance.logMessage("❌ SDK init failed: \(errorMessage)")
                }
            }
        }
    }
}
